app/services/sources/RedditSource.java

```diff
@@ -119,17 +107,34 @@ public RedditClient getRedditClient() {
         return redditClient;
     }
 
-    public List<Submission> getHotPosts() {
+    public List<Submission> getHotPosts(int numPosts) {
         SubredditPaginator sp = new SubredditPaginator(redditClient);
+        Iterator<Listing<Submission>> listingIterator = sp.iterator();
 
-        sp.setLimit(MAX_REQUEST_SIZE);
+        sp.setLimit(numPosts);
         sp.setSorting(Sorting.HOT);
         sp.setTimePeriod(TimePeriod.DAY);
         sp.next(true);
 
-        Listing<Submission> list = sp.getCurrentListing();
+        ArrayList<Submission> hotPosts = new ArrayList<>();
+        Listing<Submission> page = sp.getCurrentListing();
+
+        hotPosts.addAll(page);
+
+        //Iterate through next Reddit pages and add to hotPosts. Each page only returns 100
+        while(listingIterator.hasNext() && hotPosts.size() <= numPosts) {
+            Listing<Submission> nextPage = listingIterator.next();
 
-        return list;
+            hotPosts.addAll(nextPage);
+        }
+
+        //Only return numPosts # of hot posts from the list
+        for(int i=hotPosts.size()-1; i>=numPosts; i--) {
+            hotPosts.remove(i);
+        }
```

Comment:
can use something like [subList](http://stackoverflow.com/questions/12099721/how-to-use-sublist) here